conans/client/build/autotools_environment.py

```diff
@@ -104,21 +112,28 @@ def _get_triplet(self, the_arch, the_os):
 
         return "%s-%s" % (machine, op_system)
 
-    def _get_host_build_target_flags(self, arch_detected, os_detected):
+    def _get_host_build_target_flags(self, arch_detected=None, os_detected=None):
         """Based on google search for build/host triplets, it could need a lot
         and complex verification"""
 
-        if not cross_building(self._conanfile.settings, os_detected, arch_detected):
+        arch_platform = arch_detected or detected_architecture() or platform.machine()
```

Comment:
Due we are removing the opt parameters I would prefer "arch_detected" and "os_detected" instead of "arch_platform" and "os_platform".